src/main/webapp/app/exercises/programming/manage/update/update-components/information/programming-exercise-information.component.ts

```diff
@@ -0,0 +1,219 @@
+    }
+
+    ngOnDestroy(): void {
+        for (const subscription of this.inputFieldSubscriptions) {
+            subscription?.unsubscribe();
+        }
+    }
+
+    calculateFormValid() {
+        const isCheckoutSolutionRepositoryValid = this.isCheckoutSolutionRepositoryValid();
+        const isRecreateBuildPlansValid = this.isRecreateBuildPlansValid();
+        const isUpdateTemplateFilesValid = this.isUpdateTemplateFilesValid();
+        const areAuxiliaryRepositoriesValid = this.areAuxiliaryRepositoriesValid();
+        this.formValid = Boolean(
+            this.exerciseTitleChannelComponent()?.titleChannelNameComponent?.formValidSignal() &&
+                this.getIsShortNameFieldValid() &&
+                isCheckoutSolutionRepositoryValid &&
+                isRecreateBuildPlansValid &&
+                isUpdateTemplateFilesValid &&
+                areAuxiliaryRepositoriesValid,
+        );
+        this.formValidChanges.next(this.formValid);
+    }
+
+    areAuxiliaryRepositoriesValid(): boolean {
+        return (
+            (every(
+                this.tableEditableFields?.map((field) => field.editingInput.valid),
+                Boolean,
+            ) &&
+                !this.programmingExerciseCreationConfig.auxiliaryRepositoryDuplicateDirectories &&
+                !this.programmingExerciseCreationConfig.auxiliaryRepositoryDuplicateNames) ||
+            !this.programmingExercise().auxiliaryRepositories?.length
+        );
+    }
+
+    isUpdateTemplateFilesValid(): boolean {
+        return (
+            this.updateTemplateFilesField?.valid ||
+            !this.programmingExerciseCreationConfig.isImportFromExistingExercise ||
+            this.programmingExercise().projectType === ProjectType.PLAIN_GRADLE ||
+            this.programmingExercise().projectType === ProjectType.GRADLE_GRADLE
+        );
+    }
+
+    isRecreateBuildPlansValid(): boolean {
+        return this.recreateBuildPlansField?.valid || !this.programmingExerciseCreationConfig.isImportFromExistingExercise;
+    }
+
+    isCheckoutSolutionRepositoryValid(): boolean {
+        return Boolean(
+            this.checkoutSolutionRepositoryField?.valid ||
+                this.programmingExercise().id ||
+                !this.programmingExercise().programmingLanguage ||
+                !this.programmingExerciseCreationConfig.checkoutSolutionRepositoryAllowed,
+        );
+    }
+
+    private updateIsShortNameValid() {
+        this.isShortNameFieldValid.set(this.getIsShortNameFieldValid());
+    }
+
+    private getIsShortNameFieldValid() {
+        return this.shortNameField() === undefined || this.shortNameField()?.control?.status === 'VALID';
+    }
+
+    private ensureShortNameIsUnique(shortName: string): string {
+        let newShortName = shortName;
+        let counter = 1;
+        while (this.alreadyUsedShortNames().includes(newShortName)) {
+            if (counter > MAXIMUM_TRIES_TO_GENERATE_UNIQUE_SHORT_NAME) {
+                break;
```

Comment:
Should you really simply break the loop and return some string?
Can you instead inform the user that they have to go into the hard mode and add a unique short name themselves?

Reply:
Yeah, I do not think that this case will ever occur but it is better to display an error message to the user - I adjusted it 👍